packages/entities/entities-shared/src/components/entity-empty-state/EntityEmptyState.vue

```diff
@@ -0,0 +1,230 @@
+      >
+        <p>
+          <b>{{ t('emptyState.pricingTitle') }}</b> {{ pricing }}
+        </p>
+      </div>
+    </div>
+    <div
+      v-if="$slots.message"
+      class="entity-empty-state-message"
+    >
+      <slot name="message" />
+    </div>
+    <div class="entity-empty-state-action">
+      <KButton
+        v-if="(actionButtonText) || $slots.action"
+        appearance="primary"
+        size="large"
+        @click="$emit('create-button-clicked')"
+      >
+        <AddIcon />
+        {{ actionButtonText }}
+      </KButton>
+      <KButton
+        v-if="learnMoreLink"
+        appearance="secondary"
+        size="large"
+        @click="$emit('learning-hub-button-clicked')"
+      >
+        <BookIcon decorative />
+        {{ t('emptyState.learnMore') }}
+      </KButton>
+    </div>
+    <div class="entity-empty-state-card-container">
+      <template
+        v-for="feature in features"
+        :key="feature"
+      >
+        <KCard class="entity-empty-state-card">
+          <template #title>
+            <component
+              :is="getEntityIcon(feature.iconVariant)"
+              :color="`var(--kui-color-text-neutral-stronger, ${KUI_COLOR_TEXT_NEUTRAL_STRONGER})`"
+              :size="KUI_ICON_SIZE_40"
+            />
+            <div class="card-title">
+              {{ feature.title }}
+            </div>
+          </template>
+          {{ feature.description }}
+        </KCard>
+      </template>
+    </div>
+  </div>
+</template>
+
+<script lang="ts" setup>
+import { type PropType } from 'vue'
+import { KButton } from '@kong/kongponents'
+import { BookIcon, AddIcon, DeployIcon, PlugIcon, ChartDataIcon, AnalyticsIcon } from '@kong/icons'
+import composables from '../../composables'
+import type { EmptyStateFeature } from 'src/types/entity-empty-state'
+import { KUI_ICON_SIZE_40, KUI_COLOR_TEXT_NEUTRAL_STRONGER } from '@kong/design-tokens'
+
+const getEntityIcon = (iconType: string): object => {
+  console.log(iconType)
+  switch (iconType) {
+    case 'deploy':
+      return DeployIcon
+    case 'plug':
+      return PlugIcon
+    case 'chartData':
+      return ChartDataIcon
+    case 'analytics':
+      return AnalyticsIcon
+    default:
+      return BookIcon
+  }
+}
+
+defineProps({
+  title: {
+    type: String,
+    required: true,
+  },
+  description: {
+    type: String,
+    required: true,
+  },
+  pricing: {
+    type: String,
+    default: '',
+  },
+  actionButtonText: {
+    type: String,
+    default: '',
+  },
+  learnMoreLink: {
+    type: String,
+    default: '',
+  },
+  features: {
+    type: Array as PropType<EmptyStateFeature[]>,
+    default: () => [],
+  },
+})
+
+defineEmits(['create-button-clicked', 'learning-hub-button-clicked'])
+
+const { i18n: { t } } = composables.useI18n()
+</script>
+
+<style lang="scss" scoped>
+.kong-ui-public-entity-empty-state {
+  align-items: center;
+  background-color: var(--kui-color-background, $kui-color-background);
```

Comment:
```suggestion
  background-color: $kui-color-background;
```

suggestion: we don't need to reference CSS vars to allow overriding for theme in this component. This applies to all of these styles

Reply:
Removed all CSS vars